### PR TITLE
Get Accounts Operation for SSO extension [common core]

### PR DESCRIPTION
**Pull request by jasoncoolmax**
This is a branch which cherry picks all the changes from [this one](https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/584) and points to the right target branch.

The target branch is jason/removeAccountOperation because some of the change depends on some change in that branch. Both PRs will be merged to sso release branch.

### CODE REVIEW

File: IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetAccountsResponse.m [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        NSDictionary *responseData = json[@"response_data"];
+        
+        if (![responseData msidAssertType:NSArray.class ofKey:@"accounts" required:YES error:error])
+        {
+            return nil;
+        }
+        NSArray *accountsJson = responseData[@"accounts"];
+        
+        NSMutableArray *accounts = [NSMutableArray new];
+        for (NSDictionary *accountJson in accountsJson)
```
Comment (oldalton) [inline IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetAccountsResponse.m]: nit: should we verify accountJson is a dictionary?

File: IdentityCore/src/oauth2/account/MSIDAccount.m [hunk 2]
```diff
@@ -178,4 +178,68 @@ - (NSString *)description
...
+    {
+        return nil;
+    }
+    
+    if (![json msidAssertType:NSDictionary.class ofKey:@"account_identifier" required:YES error:error])
+    {
+        return nil;
+    }
+    
+    self.accountIdentifier = [[MSIDAccountIdentifier alloc] initWithJSONDictionary:json[@"account_identifier"] error:error];
```
Comment (oldalton) [inline IdentityCore/src/oauth2/account/MSIDAccount.m]: I thought nested dictionaries weren't allowed. @antrix1989?

Comment (antrix1989) [reply IdentityCore/src/oauth2/account/MSIDAccount.m]: correct, we need to use flat json.

File: IdentityCore/src/broker_operation/request/account_request/MSIDBrokerOperationGetAccountsRequest.m [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+- (NSDictionary *)jsonDictionary
+{
+    NSMutableDictionary *json = [[super jsonDictionary] mutableCopy];
+    if (!json) return nil;
+    
+    if (!self.clientId) return nil;
+    [json setValue:self.clientId forKey:MSID_BROKER_CLIENT_ID_KEY];
+    
+    if (self.familyId)
```
Comment (antrix1989) [inline IdentityCore/src/broker_operation/request/account_request/MSIDBrokerOperationGetAccountsRequest.m]: nit: we can do `json[forKey:MSID_BROKER_FAMILY_ID_KEY] = self.familyId,` so we can get rid of "if".

Comment (jasoncoolmax) [reply IdentityCore/src/broker_operation/request/account_request/MSIDBrokerOperationGetAccountsRequest.m]: addressed :)

File: IdentityCore/src/oauth2/account/MSIDAccount.m (hunk 2 above)
Comment (jasoncoolmax) [reply IdentityCore/src/oauth2/account/MSIDAccount.m]: addressed :)

File: IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetAccountsResponse.m (hunk 1 above)
Comment (jasoncoolmax) [reply IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetAccountsResponse.m]: addressed :)

**Pull request merged**